MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.Api.Tests/Controllers/ListControllerTests.cs

```diff
@@ -59,76 +44,391 @@ public void Init()
+            };
+
+            _cache.GetAllItemsAsync().Returns(items);
 
             var message = await _listController.ExecuteAction(controller => controller.GetAsync());
-            message.TryGetContentValue(out IEnumerable<ListItem> items);
+            message.TryGetContentValue(out IEnumerable<IListItem> resultItems);
 
-            Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
-            Assert.That(items, Is.EqualTo(_items).Using(_listItemEqualityComparer));
+            Assert.Multiple(() => { 
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(resultItems, Is.EqualTo(items).Using(ListItemEqualityComparer.Instance));
+            });
         }
 
         [Test]
-        public async Task Get_ReturnsItemWithGivenId()
+        public async Task Get_ExistingId_ReturnsItemWithGivenId()
         {
-            var expectedItem = _items[0];
-            _repository.GetItemAsync(expectedItem.Id).Returns(Task.FromResult(expectedItem));
+            var expectedItem = ListItemBuilder.CreateItem("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC", "aaaaa", "1589-12-03", "1896-04-07", true);
+            _cache.ExistsItemWithIdAsync(expectedItem.Id).Returns(true);
+            _cache.GetItemAsync(expectedItem.Id).Returns(Task.FromResult(expectedItem));
 
             var message = await _listController.ExecuteAction(controller => controller.GetAsync(expectedItem.Id));
-            message.TryGetContentValue(out ListItem item);
+            message.TryGetContentValue(out IListItem item);
 
-            Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
-            Assert.That(item, Is.EqualTo(expectedItem).Using(_listItemEqualityComparer));
+            Assert.Multiple(() => {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(item, Is.EqualTo(expectedItem).Using(ListItemEqualityComparer.Instance));
+            });
         }
 
         [Test]
-        public async Task Post_CreatedReturned()
+        public async Task Get_EmptyId_BadRequest()
         {
-            var newItem = new ListItem { Text = "newItem" };
-            var createdItem = new ListItem
-            {
-                Id = new Guid("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAB"),
-                Text = "newItem",
-                IsActive = false,
-                CreationTime = new DateTime(1589, 12, 3),
-                LastUpdateTime = new DateTime(1896, 4, 7)
-            };
+            var id = Guid.Empty;
+
+            var message = await _listController.ExecuteAction(controller => controller.GetAsync(id));
+            message.TryGetContentValue(out HttpError error);
+
+            Assert.Multiple(() => {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+                Assert.That(error.ModelState.Keys.Contains(nameof(IListItem.Id)), Is.True);
```

Comment:
```suggestion
                Assert.That(error.ModelState, Does.ContainKey(nameof(IListItem.Id)));
```
Whenever you see `Is.True` or `Is.False` it can most likely be refactored to something more expressive unless actual value is a `boolean` 😉